app/services/email_sender/email_sender_service.rb

```diff
@@ -0,0 +1,30 @@
+class EmailSenderService
+  def initialize(config, email_service_provider)
+    @email_address_override = config[:email_address_override]
+    @provider = email_service_provider
+  end
+
+  def call(address:, subject:, body:)
+    log_override(address: address, subject: subject, body: body) if @email_address_override
+
+    @provider.call(
```

Comment:
I would be tempted to make this and `email_address_override` an `attr_reader` to look a bit nicer.